Dockerfile

```
@@ -1,212 +1,58 @@
-  BUNDLE_SILENCE_ROOT_WARNING=1 \
-  BUNDLE_APP_CONFIG="$GEM_HOME"
-# path recommendation: https://github.com/bundler/bundler/pull/6469#issuecomment-383235438
-ENV PATH $GEM_HOME/bin:$BUNDLE_PATH/gems/bin:$PATH
-# adjust permissions of a few directories for running "gem install" as an arbitrary user
-RUN mkdir -p "$GEM_HOME" && chmod 777 "$GEM_HOME"
-# (BUNDLE_PATH = GEM_HOME, no need to mkdir/chown both)
-
-#------------------------------------------------------------------------------------------
+#------------------------------------------------------------------------------
 #
-# End of ruby installation
+# Install requiered bundler version
 #
-#------------------------------------------------------------------------------------------
+#------------------------------------------------------------------------------
+RUN gem install bundler:2.0.2
 
-# Install alpine equivalent for "build-essential"
+#------------------------------------------------------------------------------
 #
-# and other dependencies required by dev.to various
-# ruby dependencies (postgresql-dev, tzdata)
-RUN apk add --no-cache alpine-sdk postgresql-dev tzdata
-
-# Im installing bash, as im a bash addict (not that great with sh)
-RUN apk add bash
-
-# Let's setup the rails directory
-# (@TODO - consider a production version?)
-WORKDIR /usr/src/app
-ENV RAILS_ENV development
-
-#####################################################
+# Install Yarn
 #
-# Let's prepare the dev.to source code files
-# WITHOUT docker related files
-#
-# This allow us to modify the docker
-# entrypoint / run file without recompiling
-# the entire application
-# (especially when creating this build script =| )
-#
-# (@TODO - improve and review ignore to blacklist unneeded items)
-#
-#####################################################
+#------------------------------------------------------------------------------
+ENV PATH=/root/.yarn/bin:$PATH
+RUN apk add --no-cache --repository=http://dl-cdn.alpinelinux.org/alpine/edge/community \
+        yarn
 
+#------------------------------------------------------------------------------
 #
-# Prepare the source code and remove any unneeded files
+# Define working directory
 #
-FROM alpine-ruby-node AS source-code-repo
-
-# The workdir
+#------------------------------------------------------------------------------
 WORKDIR /usr/src/app
-# Copy source code
-COPY ./ /usr/src/app/
-# remove docker related files
-RUN rm Dockerfile && rm docker-*
 
+#------------------------------------------------------------------------------
 #
-# Does the source code build
+# Copy Gemfile and run bundle install
 #
-FROM alpine-ruby-node AS source-code-build
-
-# Copy over files
-COPY --from=source-code-repo /usr/src/app/ /usr/src/app/
-
-# Run the various installer
-RUN gem install bundler
+#------------------------------------------------------------------------------
+COPY Gemfile /usr/src/app/Gemfile
+COPY Gemfile.lock /usr/src/app/Gemfile.lock
```

This can be put in a single line, so we don't have to add another layer. 
```suggestion
COPY ./Gemfile ./Gemfile.lock ./
```